catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/FeedbackApplication.java

```diff
@@ -0,0 +1,289 @@
+            feedbackRequest.setQueryStatus(searchStatus.stream()
+                    .map(Object::toString)
+                    .collect(Collectors.joining(", ")));
+        }
+        feedbackRequest.setWorkspaceId(workspaceId);
+        feedbackRequest.setWorkspaceName(workspaceName);
+        feedbackRequest.setComments(comments);
+
+        return feedbackRequest;
+    }
+
+    private String getEmailSubject(FeedbackRequest feedback) {
+        initConfigurationAppValues(false);
+        return applyTemplate(subjectTemplate, feedback);
+    }
+
+    private String getEmailBody(FeedbackRequest feedback) {
+        initConfigurationAppValues(false);
+        return applyTemplate(bodyTemplate, feedback);
+    }
+
+    private String applyTemplate(Template template, FeedbackRequest feedback) {
+        String templatedString = null;
+
+        if (template != null) {
+            Map<String, String> templateValueMap = getTemplateValueMap(feedback);
+            try {
+                templatedString = template.apply(templateValueMap);
+            } catch (IOException e) {
+                LOGGER.error("Unable to apply values to handle bars email body template", e);
+            }
+        }
+
+        return templatedString;
+    }
+
+    private void initConfigurationAppValues(boolean reinit) {
+        Handlebars handlebars = new Handlebars();
+
+        if (reinit || subjectTemplate == null || bodyTemplate == null) {
+            if (configurationApplication != null) {
+                emailDestination = configurationApplication.getQueryFeedbackEmailDestination();
+                String subjectTemplateStr =
+                        configurationApplication.getQueryFeedbackEmailSubjectTemplate();
+                String bodyTemplateStr =
+                        configurationApplication.getQueryFeedbackEmailBodyTemplate();
+
+                try {
+                    if (subjectTemplateStr != null) {
+                        subjectTemplate = handlebars.compileInline(subjectTemplateStr);
+                    }
+
+                    if (bodyTemplateStr != null) {
+                        bodyTemplate = handlebars.compileInline(bodyTemplateStr);
+                    }
+                } catch (IOException e) {
+                    LOGGER.warn("Unable to compile email templates", e);
+                }
+            } else {
+                LOGGER.debug("Feedback configuration is not set");
+            }
+        }
+    }
+
+    private Map<String, String> getTemplateValueMap(FeedbackRequest feedbackRequest) {
+        Map<String, String> valueMap = new HashMap<>();
+
+        valueMap.put("auth_username", feedbackRequest.getAuthUsername());
+        valueMap.put("username", feedbackRequest.getUsername());
+        valueMap.put("email", feedbackRequest.getEmail());
+        valueMap.put("workspace_id", feedbackRequest.getWorkspaceId());
+        valueMap.put("workspace_name", feedbackRequest.getWorkspaceName());
```

Comment:
There's more here that needs the escapeHTML4.  Workspace name, query names, even the results themselves.